subprojects/dependency-lock/src/main/java/org/gradle/internal/dependencylock/model/DependencyVersion.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.dependencylock.model;
+
+public class DependencyVersion {
+
+    private final String requestedVersion;
```

Comment:
❌ What do we need this for?

Reply:
At the moment this is only used for test assertions e.g. `foo:bar:1.5 -> 1.5` whereas the first version in the String is the `requestedVersion`.

Reply:
The test should just assert that `foo:bar` was locked to version 1.5. We don't need the requested version. See my follow-up on the other comment about multiple versions.